### DISCUSSION TITLE
Announcement plan

### DISCUSSION THREAD

**Discussion by Shnatsel** [etemesi254/zune-image#81] [General]
How would you like to go about announcing the project?

I think announcing PNG and JPEG separately as soon as they're ready could be beneficial. Mostly because the PNG decoder would make for a great headline, and separate announcements usually drive more clicks. But you could also announce `zune-image` as a whole once it's ready, if you prefer.

I can help write effective release announcements. I have been doing plenty of that, [here's an example](https://www.reddit.com/r/rust/comments/hgrum2/png_crate_is_now_4x_faster_supports_apng/).

**Comment by Shnatsel** [comment]
Also, not all performance work has to be merged in the initial release, only correctness really matters. Performance improvements in subsequent releases make for great subsequent release announcements, driving more users to the library.

**Comment by etemesi254** [comment]
- `zune-image` is a bit far from ready, I'm still making breaking changes almost everyday as I think about accommodating more formats,(e.g we can theoretically support floating point images, natively but I haven't yet figured out how to do image operations in floating point.
Also we don't have support for animated images yet)

- `zune-jpeg` should be ready once I fix that color bug and any fuzz crashes. 

- `zune-png`, I need to think about lower bit-depths and how I will deal with them.

And finally I feel like I need to explain the end goals, i.e why it is like this and why are these changes not in image-rs, because I know someone will ask.

Other than that I feel like you can write the release announcements, I'll write 2-3 blogs with some notes, why's and end goals and we can start the release.

I can't give a real timeline, this is just a side project so I can't say exactly when

**Comment by etemesi254** [comment]
Update 

I think jpeg is almost done, I hope I haven't spoken too soon.

PNG has that issue at https://github.com/etemesi254/zune-image/issues/84 that I'm looking at, some images are animated hence why we have problems while others are genuine bugs in the decoder, but it should not be too long for it to be correct. Lower bit depths I'm still thinking about it.

**Comment by Shnatsel** [comment]
Indeed, JPEG seems to be done. I'm going to do one last fuzzing pass and then share the fuzzing seeds. The preliminary fuzzing results are good.

**Reply by etemesi254** [reply]
When satisfied with the state of it, ping me to make a release, you can still do the announcement.
